optuna/integration/wandb.py

```diff
@@ -51,15 +57,17 @@ def objective(trial):
                 x = trial.suggest_float("x", -10, 10)
                 return (x - 2) ** 2
 
+            study = optuna.create_study(study_name="my_study")
 
             wandb_kwargs = {"project": "my-project"}
-            wandbc = WeightsAndBiasesCallback(wandb_kwargs=wandb_kwargs)
-
+            wandbc = WeightsAndBiasesCallback(study_name=study.name, wandb_kwargs=wandb_kwargs)
 
-            study = optuna.create_study(study_name="my_study")
             study.optimize(objective, n_trials=10, callbacks=[wandbc])
 
     Args:
+        study_name:
+            Study name for which the Callback is being used. This is used for tagging
+            run(s) generated during the experiment.
```

Comment:
Sure.

Reply:
I kept it this way to maintain uniformity in tagging runs in both the scenarios. Also, since one can put multiple tags to a run, adding an extra tag, should cause no harm and just act as an extra piece of information in identifying runs which an user may or may not choose to use. Also, auto grouping works only when `group=` is set in `wandb.init(...)`. Setting an extra tag should be virtually harmless.

Setting this tag can also be useful in the off chance of an user creates and logs multiple studies (or a single study multiple times) to the same project. Since run names and run-ids are randomly generated, having no useful information to identify the optuna study that created the run, tags can helpful in such scenarios.

Reply:
I agree with @xadrianzetx for two reasons:

1. This change breaks the backward compatibility of the callback. Even though this callback is still experimental, ideally we would like to avoid such a change. 
2. We can do the same thing by using `wandb_kwargs`. Related to this, the default `study_name` of `create_study` is `None`, which means users do not give a name to a study.

Reply:
Sure. But doesn't optuna create a random uuid study name if `None` is provided? The idea is to tag the runs with that study name. How can I do that without asking for the study name as a parameter?

Reply:
IMO behavior would be:

* With `as_sweeps` being `False` - tagging run with study name would be purely optional and done with entry to `wandb_kwargs` specified by user. The reason is that in this mode 1 study = 1 wandb run, so even if user runs multiple studies as a part of the same project, it's obvious that runs are separate entities and it's up to user to decide how they want to distinguish between them (as @nzw0301 said, default study name might be used, which IMO does not really help to pinpoint experiment. Something like date would be more useful).
* With `as_sweeps` being `True` - here multiple wandb runs will refer to the same study, and if user runs multiple studies within the same project, we will lose track of what's what. I think name tagging can be useful here, since it solves this problem (doesn't matter what the study name is, as long as we can filter runs by it). In this mode, we are initializing new wandb run on each callback call and we have `study` instance available there, so we can use it to derive study name automatically and pass it to `wandb.init`.